runtimes/node-18.0/server.js

```diff
@@ -18,25 +18,51 @@ const server = micro(async (req, res) => {
     };
 
     const contentType = req.headers['content-type'] ?? 'text/plain';
-    const rawBody = await parseText(req);
-    let body = rawBody;
+    const bodyString = await parseText(req);
+    let body = bodyString;
 
     if (contentType.includes('application/json')) {
-        body = await parseJson(req);
+        if(bodyString) {
+            body = await parseJson(req);
+        } else {
+            body = {};
+        }
     }
 
     const headers = {};
     Object.keys(req.headers).filter((header) => !header.toLowerCase().startsWith('x-open-runtimes-')).forEach((header) => {
         headers[header.toLowerCase()] = req.headers[header];
     });
 
+    const scheme = (req.headers['x-forwarded-proto'] ?? 'http');
+    const host = req.headers['host'].includes(':') ? req.headers['host'].split(':')[0] : req.headers['host'];
+    const port = +(req.headers['host'].includes(':') ? req.headers['host'].split(':')[1] : '80');
+    const path = req.url.includes('?') ? req.url.split('?')[0] : req.url;
+    const queryString = req.url.includes('?') ? req.url.split('?')[1] : '';
```

Comment:
why are we doing all these steps manually? We can use the URL class of Node to do all this 

![Screenshot 2023-02-07 at 3 42 55 PM](https://user-images.githubusercontent.com/20852629/217216238-294a057e-3ffd-413d-bad9-c06a750eb0b0.png)

Reply:
If unit tests can cover all possible scenarios, i'm okay with it. Else its safest to go with a standard library when it exists. They're battle tested over the years

Reply:
Ill use std library.